text/0000-project-error-handling.md

```diff
@@ -43,6 +43,13 @@ Here is a tenative starting point, subject to change:
 - What is the consensus on handling `dyn Error`s? Should it be encouraged or discouraged? Should we look into making `Box<dyn Error...>` implement `Error`?
 
 
+### Identify pain points in error handling today
+
+- Backtrace capture is expensive, but without one it can be difficult to pinpoint the origin of errors
```

Comment:
```suggestion
- Backtrace capture is expensive, but without it can be difficult to pinpoint the origin of errors
```

Reply:
I generally disagree with the universal applicability of this point as well; this is a key point of SNAFU — producing a semantic backtrace via the error types.

Reply:
Assuming I understand what you mean, I think that the semantic backtraces via `SNAFU` can definitely help here but I don't think they're a universal solution. It still fairly easy to misuse. For example, if a developer introduces a higher level snafu error to handle an `io::Error`, but then uses this type liberally throughout the code. Unless you carefully create one variant per source location the origin of errors can still be lost.

Here's an example of this kind of thing going wrong via `thiserror`. https://github.com/ZcashFoundation/zebra/issues/758